RepoMan/RepoMan/Repository/BitBucketCloudPullRequestReader.cs

```diff
@@ -0,0 +1,359 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using RepoMan.Records;
+using ILogger = Serilog.ILogger;
+
+namespace RepoMan.Repository
+{
+    public class BitBucketCloudPullRequestReader :
+        IRepoPullRequestReader
+    {
+        private readonly string _prApiUrl;
+        private readonly string _repoTag;
+        private readonly HttpClient _bbClient;
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+        private readonly IClock _clock;
+        private readonly ILogger _logger;
+
+        public BitBucketCloudPullRequestReader(string hostname, string repoOwner, string repoName, HttpClient client, JsonSerializerSettings jsonSerializerSettings, IClock clock, ILogger logger)
+        {
+            if (!Uri.TryCreate(hostname, UriKind.Absolute, out var _)) throw new ArgumentException($"'{hostname}' is not a valid URL");
+            if (string.IsNullOrWhiteSpace(repoOwner)) throw new ArgumentNullException(nameof(repoOwner));
+            if (string.IsNullOrWhiteSpace(repoName)) throw new ArgumentNullException(nameof(repoName));
+            _repoTag = $"{repoOwner}:{repoName}";
+            _jsonSerializerSettings = jsonSerializerSettings ?? throw new ArgumentNullException(nameof(jsonSerializerSettings));
+            _clock = clock ?? throw new ArgumentNullException(nameof(clock));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _bbClient = client ?? throw new ArgumentNullException(nameof(client));
+
+            var rawHostname = string.IsNullOrWhiteSpace(hostname)
+                ? throw new ArgumentNullException(nameof(hostname))
+                : hostname;
+            
+            const string apiSlug = "!api/2.0";
+
+            var normalizedApiHost = rawHostname.EndsWith("/", StringComparison.Ordinal)
```

Comment:
Uri handles this for you? why not actually use the out from above?

Reply:
Hmm, that's a good idea.

Reply:
Fixed